torch/_dynamo/allowed_functions.py

```diff
@@ -315,4 +321,4 @@ def is_builtin_constant(obj):
 
 
 def is_numpy(obj):
-    return isinstance(obj, np.ndarray) or id(obj) in _numpy_function_ids
+    return HAS_NUMPY and (isinstance(obj, np.ndarray) or id(obj) in _numpy_function_ids)
```

Comment:
Nit: this way you don't need to introduce HAS_NUMPY (it feels a bit odd to define both to be frank)
```suggestion
    return (isinstance(obj, np.ndarray)  if np is not None else False) or id(obj) in _numpy_function_ids
```